src/pathfinder/yapf/yapf_destrail.hpp

```diff
@@ -166,16 +166,13 @@ class CYapfDestinationTileOrStationRailT : public CYapfDestinationRailBase {
 	/** Called by YAPF to detect if node ends in the desired destination */
 	inline bool PfDetectDestination(TileIndex tile, Trackdir td)
 	{
-		bool bDest;
 		if (m_dest_station_id != INVALID_STATION) {
-			bDest = HasStationTileRail(tile)
+			return HasStationTileRail(tile)
 				&& (GetStationIndex(tile) == m_dest_station_id)
 				&& (GetRailStationTrack(tile) == TrackdirToTrack(td));
-		} else {
-			bDest = (tile == m_destTile)
-				&& ((m_destTrackdirs & TrackdirToTrackdirBits(td)) != TRACKDIR_BIT_NONE);
 		}
-		return bDest;
+
+		return (tile == m_destTile) && ((m_destTrackdirs & TrackdirToTrackdirBits(td)) != TRACKDIR_BIT_NONE);
```

Comment:
I think this change decreases readability.

Reply:
Disagree here - the condition is unchanged, it's just moved to an early return style instead of a boolean, which ultimately reduces the length of the code.

Reply:
Also don't see why's that a decrease. Though I would probably switch the if to check for INVALID_STATION corner case and leave general result as a main body.